src/main/java/com/aws/greengrass/util/BatchedSubscriber.java

```diff
@@ -0,0 +1,148 @@
+     */
+    public static BatchedSubscriber subscribe(Topic topic, Runnable afterBatch) {
+        return subscribe(topic, afterBatch, null, null);
+    }
+
+    /**
+     * Subscribe to topic changes. Subscriptions will only be called once
+     * for a batch of changes.
+     *
+     * @param topic            topic
+     * @param afterBatch       callback to perform after a batch of changes
+     * @param onInitialization callback to perform after the subscription has been added
+     * @param excludeFilter    custom filter for ignoring changes
+     * @return batched subscriber
+     */
+    public static BatchedSubscriber subscribe(Topic topic,
+                                              Runnable afterBatch, Runnable onInitialization,
+                                              BiPredicate<WhatHappened, Topic> excludeFilter) {
+        AtomicInteger numRequestedChanges = new AtomicInteger();
+        AtomicInteger totalCalls = new AtomicInteger();
+        Subscriber subscriber = (what, child) -> {
+            if (what == WhatHappened.initialized) {
+                if (onInitialization != null) {
+                    onInitialization.run();
+                }
+                return;
+            }
+
+            if (what == WhatHappened.timestampUpdated || what == WhatHappened.interiorAdded) {
+                return;
+            }
+
+            if (excludeFilter != null && excludeFilter.test(what, child)) {
+                return;
+            }
+
+            numRequestedChanges.incrementAndGet();
+            topic.context.runOnPublishQueue(() -> {
+                if (numRequestedChanges.decrementAndGet() == 0) {
+                    afterBatch.run();
+                }
+            });
+        };
+        topic.subscribe(subscriber);
+        return new BatchedSubscriber(() -> topic.remove(subscriber));
+    }
+
+    /**
+     * Subscribe to topic changes. Subscriptions will only be called once
+     * for a batch of changes.
+     *
+     * @param topics     topics
+     * @param afterBatch callback to perform after a batch of changes
+     * @return batched subscriber
+     */
+    public static BatchedSubscriber subscribe(Topics topics, Runnable afterBatch) {
+        return subscribe(topics, afterBatch, null, null);
+    }
+
+    /**
+     * Subscribe to topic changes. Subscriptions will only be called once
+     * for a batch of changes.
+     *
+     * @param topics           topics
+     * @param afterBatch       callback to perform after a batch of changes
+     * @param onInitialization callback to perform after the subscription has been added
+     * @param excludeFilter    custom filter for ignoring changes
+     * @return batched subscriber
+     */
+    public static BatchedSubscriber subscribe(Topics topics,
+                                              Runnable afterBatch, Runnable onInitialization,
+                                              BiPredicate<WhatHappened, Node> excludeFilter) {
```

Comment:
I'd rather not split out the batch and initialization callbacks. Combine them and pass into the callback either WhatHappened.initialized or WhatHappened.changed accordingly.

Also, rather than a predicate for filtering I'd suggest a simple `WhatHappened[]` of exclusions.

Reply:
To provide some context to `excludeFilter`: the use-case I had in mind was to ignore a subset of config changes:
```
// treat mqttTopicMapping property changes differently than rest
var excludeFilter = (what, child) -> child.childOf("mqttTopicMapping");
```
and I threw in `WhatHappened` in the filter so that caller could have maximum flexibility on the subscription.


Thoughts on having a `Predicate<Node>` for filtering in addition to WhatHappened[] exclusions?

Reply:
Your current implementation is fine for that yes. Don't add the array